### PR TITLE
Fix a nullptr crash.

### PR DESCRIPTION
**Pull request by TroutZhang**
Also, add a warning log to remind people to re-save data for this situation.

### CODE REVIEW

**Comment by TroutZhang** [comment]
Original crash callstack:

>	[Inline Frame] UnrealEditor-VisualStudioTools.dll!VSTools::FindBlueprintNativeParents(const UClass *) Line 74	C++	Symbols loaded.
 	UnrealEditor-VisualStudioTools.dll!VSTools::AssetIndex::ProcessBlueprint(**const UBlueprintGeneratedClass * BPGC=0x0000000000000000**) Line 119	C++	Symbols loaded.
 	UnrealEditor-VisualStudioTools.dll!VSTools::ProcessAssets(VSTools::AssetIndex & Index={...}, const TArray<FAssetData,TSizedDefaultAllocator<32>> & TargetAssets={...}) Line 420	C++	Symbols loaded.
 	UnrealEditor-VisualStudioTools.dll!VSTools::RunAssetScan(VSTools::AssetIndex & Index={...}, const TArray<TWeakObjectPtr<UClass,FWeakObjectPtr>,TSizedDefaultAllocator<32>> & FilterBaseClasses={...}) Line 445	C++	Symbols loaded.
 	UnrealEditor-VisualStudioTools.dll!UVisualStudioToolsCommandlet::Main(const FString & Params) Line 562	C++	Symbols loaded.
 	UnrealEditor.exe!FEngineLoop::PreInitPostStartupScreen(const wchar_t * CmdLine=0x0000000000000000) Line 3659	C++	Symbols loaded.

**Comment by TroutZhang** [comment]
The warning log will be like this:
image

**Comment by TroutZhang** [comment]
The issue for the assets themselves is easy to fix: locate them in the editor using ObjectPath, then re-save.

**Review by hnrqbaggio — DISMISSED** [review]
LGTM.
Thank you for the contribution, @TroutZhang!

**Review by hnrqbaggio — CHANGES_REQUESTED** [review]
Hi @TroutZhang,
Do you think it would be possible to incorporate the changes that @kihira suggested to make the fix compatible with UE5.1?

The extra information you added would be useful for customers to understand why some assets are being ignored and it's something we can extend for other error cases later.

File: Source/VisualStudioTools/Private/VisualStudioToolsCommandlet.cpp [hunk 1]
```diff
@@ -417,7 +417,20 @@ static void ProcessAssets(
...
 		}
 
-		Index.ProcessBlueprint(Cast<UBlueprintGeneratedClass>(Handle->GetLoadedAsset()));
+		if (auto BPGC = Cast<UBlueprintGeneratedClass>(Handle->GetLoadedAsset()))
+		{
+			Index.ProcessBlueprint(BPGC);
+		}
+		else
+		{
+			if (!GenClassPath.ToString().Contains(TargetAssets[i].ObjectPath.ToString()))
```
[the remark this answers is not in the document]
Comment (TroutZhang) [reply Source/VisualStudioTools/Private/VisualStudioToolsCommandlet.cpp:426]: API deprecation warning C4996 won't be treated as an error in Unreal, thus this escaped my eyes last time.
Thanks for your catch, I'll make a fix.

**Comment by TroutZhang** [comment]
> Hi @TroutZhang, Do you think it would be possible to incorporate the changes that @kihira suggested to make the fix compatible with UE5.1?
> 
> The extra information you added would be useful for customers to understand why some assets are being ignored and it's something we can extend for other error cases later.

Of course, I'll submit a fix.

**Comment by TroutZhang** [comment]
@microsoft-github-policy-service agree

**Review by hnrqbaggio — APPROVED** [review]
LGTM

**Pull request merged**